// gis-phpmailer/index.js
'use strict';

const API_URL = 'http://3.104.207.15:5000/';

function fetch_post(url,data){
	return	fetch(url, {
		method: 'POST',
		headers: {
		'Content-Type': 'application/json',
		},
		body: JSON.stringify(data),
	})
	.then(response => response.json())
	.then(data => {
		return data;
	})
	.catch((error) => {
		return error;
	});
}

// to send enquiry email
exports.EnquiryEmail = async (data) => {
	let result = [];
	let msg = [];
	
	if(data['first_name'] == '')
		msg['first_name'] = 'First Name Required';
	if(data['email'] == '')
		msg['email'] = 'Email Required';
	if(data['phone'] == '')
		msg['phone'] = 'Phone Required';
	if(data['description'] == '')
		msg['description'] = 'Description Required';
	if(data['g-recaptcha-token'] != ''){
		let recaptchaData = [];
		recaptchaData["secret"] = "6LeivcwZAAAAAIuW0-f5Ogi_AgANaRgzToTddMrF";
		recaptchaData["response"] = data['g-recaptcha-token'];
		recaptchaData["remoteip"] = "";
		$.getJSON('https://api.ipify.org?format=jsonp&callback=?', function(data) {
			console.log(JSON.stringify(data, null, 2));
			recaptchaData["remoteip"] = data['ip'];
		});

		let captchaData = [];
		let recaptcha = fetch_post('https://www.google.com/recaptcha/api/siteverify',recaptchaData);
		recaptcha.then(user => {
			captchaData = user;
		})
		console.log(captchaData)
	}

	let msgLength = Object.keys(msg).length;
	if(!msgLength){
		result['status'] = true;

		let userData = fetch_post(API_URL +'users',data);
		userData.then(user => {
			result['message'] = user;
		})
		result['message'] = userData;

	}else{
		result['status'] = false;
		result['message'] = msg;
	}

	return result;
}
